Feature/timeline

MrSliddes opened the pull request:
Merge timeline into main

MrSliddes committed:
Revert "Merge pull request #22 from Amsterdam/feature/FileImporter"

This reverts commit 895bfbffa7da387a441d2e04991e3519ff5d4a1d, reversing
changes made to 465a5c6096c5cff4ee35d119d204597198f3156f.

MrSliddes committed:
Fixed Timeline Package

MrSliddes committed:
Merge branch 'feature/TimelineFix1' into feature/Timeline

MrSliddes committed:
Added TMP + Pacakge Json Data

MrSliddes committed:
Re-added File Importer Package

MrSliddes committed:
Added Files (see desc)

+ Documentation
+ Standard Timeline UI Prefab
+ Changelog
+ Readme

MrSliddes committed:
Update CSV

MrSliddes committed:
Delete ~$TimelineTestCSV.csv.meta

MrSliddes committed:
Added CSV Timeline File Importer

MrSliddes committed:
Merge branch 'main' into feature/Timeline

MrSliddes committed:
Merged Main, Fixed Bugs, Reworked Time Period

MrSliddes committed:
Fixed Loading Data bug

MrSliddes committed:
Pre Main Merge

MrSliddes committed:
Merge branch 'main' into feature/Timeline

MrSliddes committed:
Implemented Playback Speed

MrSliddes committed:
Update UI

MrSliddes committed:
Update UI

MrSliddes committed:
Updated UI Look

MrSliddes committed:
Fixed Bugs, Added x5 x10 zoom

MrSliddes committed:
Updated TimeScrubber

Made it more 'snappier' to datetimes

MrSliddes committed:
Update ScrollRectLink.cs

MrSliddes committed:
Implemented Timeline Skips

MrSliddes committed:
Added  Clear Data / Import Data

MrSliddes committed:
Moved Prefab File To Samples

MrSliddes committed:
Added Import Button

MrSliddes committed:
Hide Sample Folder

MrSliddes committed:
Merge branch 'main' into feature/Timeline

MrSliddes committed:
Timeline Package NL3D Info Samples

sambaas commented on Packages/Netherlands3D/Timeline/Runtime/Scripts/ScrollRectLink.cs:
```diff
@@ -0,0 +1,121 @@
+        private void MouseUpdate()
+        {
+            //Check if the left Mouse button is clicked
+            if(Input.GetKeyDown(KeyCode.Mouse0))
```
Ik denk dat je hier misschien beter de interface(s) kan gebruiken zoals IPointerDownHandler en IPointerUpHandler omdat het dan ook compatible is met het nieuwe InputSystem 🤔

sambaas replied:
( Als Input Handling op Both staat gaat dit goed, maar niet wanneer alleen InputSystem wordt gebruikt )

MrSliddes replied:
Heb een nieuw verbeterd script aangemaakt genaamd ScrollRectLinker.cs die alleen van de IPointerHandlers gebruik maakt

sambaas commented on Packages/Netherlands3D/Timeline/Runtime/Scripts/LayerUI.cs:
```diff
@@ -0,0 +1,63 @@
+        }
+
+        /// <summary>
+        /// Toggle the visability of the layer
+        /// </summary>
+        public void ToggleVisability()
```
typo, is 'visibility'

MrSliddes replied:
Fixed

sambaas commented on Packages/Netherlands3D/Timeline/Runtime/Scripts/RectTransformExtensions.cs:
```diff
@@ -0,0 +1,44 @@
+    /// <summary>
+    /// Extension class for RectTransform
+    /// </summary>
+    public static class RectTransformExtensions
```
Nice, erg handig! 
-,misschien dat deze beter bij de andere Extention scripts kan staan in de Core?
Core/Runtime/Scripts/<>

MrSliddes replied:
Heb hem verplaast naar de core in de extention folder

sambaas commented on Packages/Netherlands3D/Timeline/Runtime/Scripts/TimeBarInteraction.cs:
```diff
@@ -0,0 +1,92 @@
+        /// User presses mouse down on time bar
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnPointerDown(PointerEventData eventData)
+        {
+            isDragging = true;
+            mouseDownPosition = Input.mousePosition;
+            print("yo");
```
eventData heeft een position (eventData.position), die kan je beter gebruiken dan Input.mousePosition. Dan is het EventSystem degene die bepaald wat het pointer device is

MrSliddes replied:
Dit was nog een oud script dat niet meer gebruikt werd dus heb het verwijderd

MrSliddes committed:
Pull Request Fix

- Fixed prefab null reference
- Improved ScrollRectLink.cs -> ScrollRectLinker.cs
- Fixed typo LayerUI visibility
- Moved RectTransformExtentions to Core/scripts
- Removed old obselete TimeBarInteraction.cs
